docs/examples/plotting.ipynb

```diff
@@ -0,0 +1,153 @@
+   "id": "fabulous-watson",
+   "metadata": {},
+   "source": [
+    "# plotting quantified data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "silver-coordinator",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import xarray as xr\n",
+    "import pint_xarray"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dutch-wrestling",
+   "metadata": {},
+   "source": [
+    "## load the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "reserved-genetics",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds = xr.tutorial.open_dataset(\"air_temperature\")\n",
+    "data = ds.air\n",
+    "data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "moderate-comfort",
+   "metadata": {},
+   "source": [
+    "## convert units into a format understood by pint\n",
+    "\n",
+    "<div class=\"alert alert-info\">\n",
+    "    <strong>Note:</strong> this example uses the data provided by the <code>xarray.tutorial</code> functions.\n",
+    "    As such, the units attributes follow the CF conventions, which by default <code>pint</code> does not \n",
+    "    understand. To work around that, we are modifying the <code>units</code> attributes here, but in general it\n",
+    "    is better to use a library that modifies the <code>UnitRegistry</code>.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "passing-macintosh",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data.lat.attrs[\"units\"] = \"degree\"\n",
+    "data.lon.attrs[\"units\"] = \"degree\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "legal-arctic",
+   "metadata": {},
+   "source": [
+    "## quantify the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "advanced-seating",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "quantified = data.pint.quantify()\n",
+    "quantified"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "smoking-playback",
+   "metadata": {},
+   "source": [
+    "## work with the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "satellite-performer",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "monthly_means = (\n",
+    "    quantified\n",
+    "    .pint.to(\"degC\")\n",
+    "    .sel(time=\"2013\")\n",
+    "    .groupby(\"time.month\").mean()\n",
+    ")\n",
+    "monthly_means"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "temporal-joint",
+   "metadata": {},
+   "source": [
+    "## plot\n",
+    "\n",
+    "`xarray`'s plotting functions will cast the data to `numpy.ndarray`, so we need to \"dequantify\" first."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "sensitive-television",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "monthly_means.pint.dequantify(format=\"~P\").plot.imshow(col=\"month\", col_wrap=4)"
```

Comment:
I think it would be totally fine to modify `xarray.plot.utils.label_from_attrs` to do this for pint arrays with some reasonable-default choice of format.

Reply:
indeed. However, I would like to keep the tight coupling to a minimum. Maybe using hooks or entrypoints? So for now, I think asking people to `dequantify` before plotting / saving the data is fine.

Reply:
@keewis I feel like this is an example of ["you ain't gonna need it"](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). I would be in favour of special-casing `xarray.plot.utils.label_from_attrs` until such a time that we actually need an entrypoint for other array libraries. After all, xarray's plotting methods already special-case dask arrays by calling `.compute()`.

Reply:
The issue with hard-coding is that we do expect more duck arrays in the future (e.g. `pytorch` and maybe `uncertainties`), and adding special cases for all of them would quickly become impossible.

I *was* hoping to make use of `matplotlib.units` for this. For example, using:
```python
In [9]: ureg = pint.UnitRegistry()
   ...: ureg.setup_matplotlib()
   ...:
   ...: t = ureg.Quantity(np.arange(10), "s")
   ...: v = ureg.Quantity(5, "m / s")
   ...: x = v * t
   ...: 
   ...: fig, ax = plt.subplots(1, 1)
   ...: ax.plot(t, x)
   ...: 
   ...: plt.show()
```
would be pretty nice. However, while investigating #91 I found that right now the `xarray` plotting code (sometimes? always?) converts to `np.ma.masked_array` so we can't use this, yet. Additionally, this would be `matplotlib` specific, so once we add the plotting entrypoints we would need to find something new.

Which makes me wonder: should we introduce hooks based on the type of `.data` to prepare `xarray` objects for plotting? I think these hooks could be used to call  `cupy.Array.get`, `sparse.COO.todense` or `da.Array.compute` before plotting, and because the argument to the hook would be a `xarray` object we could also have the hook for `pint` call `.pint.dequantify`. I'll open a new issue on the `xarray` issue tracker to see if I'm missing anything.

Reply:
> The issue with hard-coding is that we do expect more duck arrays in the future (e.g. pytorch and maybe uncertainties), and adding special cases for all of them would quickly become impossible.

I know, I'm not suggesting we special-case every combination forever, what I'm saying is that we don't always need to make something totally general right at the start. If we can make this work for pint and dask quickly I think we should, and worry about entrypoints when other duck-array libraries get to a similar point in their integration. We're not introducing much technical debt because this would be a small change for us and no external API change for the user (we will always want `.plot()` to work bare, however it is implemented behind the scenes).

> I was hoping to make use of matplotlib.units for this.

This would be pretty nice. But again, I think we should make what we have work, then improve it later.

> Which makes me wonder: should we introduce hooks based on the type of .data to prepare xarray objects for plotting?

I think this is a good idea for a long-term solution, which deserves it's own issue.

Reply:
> Which makes me wonder: should we introduce hooks based on the type of .data to prepare xarray objects for plotting? I think these hooks could be used to call cupy.Array.get, sparse.COO.todense or da.Array.compute before plotting,

This pretty much already happens in `.values` [here](https://github.com/pydata/xarray/blob/c472f8a4c79f872edb9dcd7825f786ecb9aff5c0/xarray/core/variable.py#L262)

```python
    data = data.get() if isinstance(data, cupy_array_type) else np.asarray(data)
```

it's just hardcoded instead of an entrypoint.

Reply:
I've implemented this in https://github.com/pydata/xarray/pull/5561